assets/js/extensions/shipping-methods/pickup-location/general-settings.tsx

```diff
@@ -0,0 +1,148 @@
+				'woo-gutenberg-products-block'
+			) }
+		</p>
+		<ExternalLink
+			href={ `${ ADMIN_URL }post.php?post=${ CHECKOUT_PAGE_ID }&action=edit` }
+		>
+			{ __( 'View checkout page', 'woo-gutenberg-products-block' ) }
+		</ExternalLink>
+	</>
+);
+
+const GeneralSettings = () => {
+	const { settings, setSettingField } = useSettingsContext();
+	const [ showCosts, setShowCosts ] = useState( !! settings.cost );
+
+	return (
+		<SettingsSection Description={ GeneralSettingsDescription }>
+			<SettingsCard>
+				<CheckboxControl
+					checked={ settings.enabled }
+					name="local_pickup_enabled"
+					onChange={ setSettingField( 'enabled' ) }
+					label={ __(
+						'Enable Local Pickup',
+						'woo-gutenberg-products-block'
+					) }
+					help={ __(
+						'When enabled, local pickup will appear as an option on the block based checkout.',
+						'woo-gutenberg-products-block'
+					) }
+				/>
+				<TextControl
+					label={ __( 'Title', 'woo-gutenberg-products-block' ) }
+					name="local_pickup_title"
+					help={ __(
+						'This is the shipping method title shown to customers.',
+						'woo-gutenberg-products-block'
+					) }
+					placeholder={ __(
+						'Local Pickup',
+						'woo-gutenberg-products-block'
+					) }
+					value={ settings.title }
+					onChange={ setSettingField( 'title' ) }
+					disabled={ false }
+					autoComplete="off"
+				/>
+				<CheckboxControl
+					checked={ showCosts }
+					onChange={ () => {
+						setShowCosts( ! showCosts );
+						setSettingField( 'cost' )( '' );
+					} }
+					label={ __(
+						'Add a price for customers who choose local pickup',
+						'woo-gutenberg-products-block'
+					) }
+					help={ __(
+						'By default, the local pickup shipping method is free.',
+						'woo-gutenberg-products-block'
+					) }
+				/>
+				{ showCosts ? (
+					<>
+						<TextControl
+							label={ __(
+								'Cost',
+								'woo-gutenberg-products-block'
+							) }
+							name="local_pickup_cost"
+							help={ __(
+								'Optional cost to charge for local pickup.',
+								'woo-gutenberg-products-block'
+							) }
+							placeholder={ __(
+								'Free',
+								'woo-gutenberg-products-block'
+							) }
+							type="number"
+							value={ settings.cost }
+							onChange={ setSettingField( 'cost' ) }
+							disabled={ false }
+							autoComplete="off"
+						/>
+						<SelectControl
+							label={ __(
+								'Taxes',
+								'woo-gutenberg-products-block'
+							) }
+							name="local_pickup_tax_status"
+							help={ __(
+								'If a cost is defined, this controls if taxes are applied to that cost.',
+								'woo-gutenberg-products-block'
+							) }
+							options={ [
+								{
+									label: __(
+										'Taxable',
+										'woo-gutenberg-products-block'
+									),
+									value: 'taxable',
+								},
+								{
+									label: __(
+										'Not taxable',
+										'woo-gutenberg-products-block'
+									),
+									value: 'none',
+								},
+							] }
```

Comment:
This is not extensible by WooCommerce right? It's always those 2 values? I saw in some places that a third value is "shipping only" but it's not relevant for shipping methods.

Reply:
Correct. This is copy pasted from other core shipping methods.